constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java

```diff
@@ -3107,4 +3107,202 @@ public String deleteFacetOptionConfiguration(
+        } catch (Exception exception) {
+            throw new ConstructorException(exception);
+        }
+    }
+
+    /**
+     * Deletes sort options
+     *
+     * @param sortBy the sort by field
+     * @param sortOrder the sort order (ascending or descending)
+     * @param section the section to which the sort option belongs
+     * @return returns the deleted sort options as JSON string
+     * @throws ConstructorException if the request is invalid
+     */
+    public String deleteSortOptions(String sortBy, String sortOrder, String section)
+            throws ConstructorException {
+        if (sortBy == null || sortBy.trim().isEmpty()) {
+            throw new IllegalArgumentException("sortBy is required");
+        }
+        if (sortOrder == null || sortOrder.trim().isEmpty()) {
+            throw new IllegalArgumentException("sortOrder is required");
+        }
+
+        try {
+            HttpUrl url =
+                    this.makeUrl(Arrays.asList("v1", "sort_options"))
+                            .newBuilder()
+                            .addQueryParameter("section", section)
+                            .build();
+
+            Map<String, Object> sortOptionParams = new HashMap<String, Object>();
+            sortOptionParams.put("sort_by", sortBy);
+            sortOptionParams.put("sort_order", sortOrder);
+
+            Map<String, Object> bodyParams = new HashMap<String, Object>();
+            bodyParams.put("sort_options", Arrays.asList(sortOptionParams));
+
+            String jsonParams = new Gson().toJson(bodyParams);
+            RequestBody body =
+                    RequestBody.create(
+                            jsonParams, MediaType.parse("application/json; charset=utf-8"));
+            Request request = this.makeAuthorizedRequestBuilder().url(url).delete(body).build();
+
+            Response response = client.newCall(request).execute();
+
+            return getResponseBody(response);
+        } catch (Exception exception) {
+            throw new ConstructorException(exception);
+        }
+    }
+
+    /**
+     * Deletes sort options with default section "Products"
+     *
+     * @param sortBy the sort by field
+     * @param sortOrder the sort order (ascending or descending)
+     * @return returns the deleted sort options as JSON string
+     * @throws ConstructorException if the request is invalid
+     */
+    public String deleteSortOptions(String sortBy, String sortOrder) throws ConstructorException {
+        return deleteSortOptions(sortBy, sortOrder, "Products");
+    }
+
+    /**
+     * Retrieves all sort options
+     *
+     * @param section the section to retrieve sort options for
+     * @param sortBy optional filter by sort by field (can be null)
+     * @return returns the sort options response
+     * @throws ConstructorException if the request is invalid
+     */
+    public SortOptionsResponse retrieveSortOptions(String section, String sortBy)
```

Comment:
Rather than having this overload, we would probably prefer having an overload that takes in a `RetrieveSortOptionsRequest` so that we cover all the different request parameters.